sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -659,13 +660,19 @@ case class StructsToJson(
         (arr: Any) =>
           gen.write(arr.asInstanceOf[ArrayData])
           getAndReset()
+      case MapType(_: DataType, _: StructType, _: Boolean) =>
+        (map: Any) =>
+          val mapType = child.dataType.asInstanceOf[MapType]
+          gen.write(map.asInstanceOf[MapData], mapType)
+          getAndReset()
     }
   }
 
   override def dataType: DataType = StringType
 
   override def checkInputDataTypes(): TypeCheckResult = child.dataType match {
-    case _: StructType | ArrayType(_: StructType, _) =>
+    case _: StructType | ArrayType(_: StructType, _) |
```

Comment:
If we support `MapType` in general, I think `ArrayType(MapType)` will be allowed? We don't need to address it in this PR.